nextcord/application_command.py

```diff
@@ -1378,6 +1404,30 @@ def verify(self) -> bool:
                 "min_value or max_value can only be set if the type is integer or number."
             )
 
+        if self.min_length is not None and not isinstance(self.min_length, int):
+            raise ValueError("min_length must be an int.")
+
+        if self.max_length is not None and not isinstance(self.max_length, int):
+            raise ValueError("max_length must be an int.")
+
+        if self.min_length is not None and self.min_length < 0:
+            raise ValueError("min_length must be greater than or equal to 0.")
+
+        if self.max_length is not None and self.max_length < 1:
+            raise ValueError("max_length must be greater than or equal to 1.")
+
+        # we check this ourselves because Discord doesn't do it yet
+        # see here: https://github.com/discord/discord-api-docs/issues/5149
+        if (
+            self.min_length is not None and self.max_length is not None
+        ) and self.min_length > self.max_length:
+            raise ValueError("min_length must be less than or equal to max_length")
+
+        if (self.min_length is not None or self.max_length is not None) and self.type is not (
+            ApplicationCommandOptionType.string
+        ):
+            raise ValueError("min_length or max_length can only be set if the type is a string.")
```

Comment:
I believe for the most part Nextcord is avoiding adding manual constraint checks and letting Discord HTTP exceptions handle most issues.

Reply:
I know, but the if statement with the comments is necessary because of that issue with the Discord API
When the issue is ever resolved then I could remove that check, but for now, that won't be the case

Reply:
If Discord allows it, does it need to be checked? While it might not function in a friendly way, the developer should be able to notice and correct the mistake while testing.

I think as a API wrapper, it should allow passing to Discord any values you want even if it doesn't make sense.

I'm open to hearing additional opinions, though.